src/PersonalityQuiz.tsx

```diff
@@ -0,0 +1,381 @@
+                `}
+            >
+                <Button
+                    type="submit"
+                    onClick={onSubmit}
+                    onKeyDown={(e) => {
+                        const spaceKey = 32;
+                        const enterKey = 13;
+                        if (e.keyCode === spaceKey || e.keyCode === enterKey)
+                            onSubmit(e);
+                    }}
+                    data-testid="submit-quiz"
+                >
+                    Submit
+                </Button>
+                <Button
+                    onClick={() => {
+                        setSelectedAnswers({});
+                        setHasSubmittedAnswers(false);
+                        setTopSelectedResult(null);
+                    }}
+                    onKeyDown={(e) => {
+                        const spaceKey = 32;
+                        const enterKey = 13;
+                        if (e.keyCode === spaceKey || e.keyCode === enterKey) {
+                            setSelectedAnswers({});
+                            setHasSubmittedAnswers(false);
+                            setTopSelectedResult(null);
+                        }
+                    }}
+                    data-testid="reset-quiz"
+                >
+                    Reset
+                </Button>
+            </div>
+        </form>
+    );
+};
+
+type PersonalityQuizAnswersProps = {
+    id: string;
+    number: number;
+    text: string;
+    imageUrl?: string;
+    answers: AnswerType[];
+    updateSelectedAnswer: (selectedAnswerId: string) => void;
+    selectedAnswer?: string;
+    hasSubmittedAnswers: boolean;
+};
+
+const PersonalityQuizAnswers = ({
+    id: questionId,
+    number,
+    text,
+    imageUrl,
+    answers,
+    updateSelectedAnswer,
+    selectedAnswer,
+    hasSubmittedAnswers,
+}: PersonalityQuizAnswersProps) => (
+    <fieldset className={answersWrapperStyle}>
+        <div>
+            <legend
+                className={css`
+                    margin-bottom: 12px;
+                `}
+            >
+                <span
+                    className={css`
+                        padding-right: 12px;
+                    `}
+                >
+                    {number + '.'}
+                </span>
+                {text}
+            </legend>
+        </div>
+        {imageUrl && (
+            <img
+                className={css`
+                    width: 100%;
+                `}
+                src={imageUrl}
+            />
+        )}
+        <div
+            className={cx(
+                radioButtonWrapperStyles,
+                css`
+                    label {
+                        :hover {
+                            background-color: ${hasSubmittedAnswers
+                                ? neutral[97]
+                                : neutral[86]};
+                        }
+                        /* TODO: apply same styles on focus (requires source update) */
```

Comment:
Having looked at source I dnt think there is an easy way to apply background styles when the input is on focus (this is mainly because you cannot apply styles to parent DOM elements from child).

This kind of change would probably require a rearchitecture of how Radio buttons are done. Knowing we are already overrriding current styles, there is probably a larger question on how and if we should ever override styles?

The hover effect I added was to try get the quiz to feel more like the one on frontend, and I was just able to get that effect to work on focus. Do I instead remove this effect all together?